cmake/macros/macro_check_compiler_setup.cmake

```diff
@@ -79,6 +79,9 @@ macro(check_compiler_setup _compiler_flags _linker_flags _var)
     endif()
   endforeach()
 
+  if(DEAL_II_MSVC)
+    set(CMAKE_TRY_COMPILE_CONFIGURATION ${build})
```

Comment:
`build` isn't defined here. I would rather move this line to `cmake/setup_compiler_flags.cmake` before calling `check_compiler_setup` and undefine `CMAKE_TRY_COMPILE_CONFIGURATION` after that call.

Reply:
Thanks for pointing that out. 
Original thought was to move it outside as ${build} is passed into, while I found out that the macro is used in three other different cmake files which makes that approach seems to be a little ugly.

Reply:
Yes, but this is even worse in my opinion. You could make the build type an additional argument to the macro. You will also find that this macro is used in `cmake/modules/FindDEAL_II_MPI.cmake` that doesn't use a particular build type. So you would need to pass a valid value there as well.
Ultimately, the build type shouldn't matter much for this macro anyway and it might be OK to just pick , say, `Debug` if that was enabled as possible configuration type.